src/Mongolid/DataMapper/DataMapper.php

```diff
@@ -0,0 +1,287 @@
+     * the given _id didn't exists.
+     *
+     * @param  mixed $object The object used in the operation.
+     *
+     * @return boolean Success (but always false if write concern is Unacknowledged)
+     */
+    public function update($object)
+    {
+        if (! $object->_id) {
+            $object->_id = new ObjectID;
+        }
+
+        $data = $this->parseToDocument($object);
+
+        $result = $this->getCollection()->updateOne(
+            ['_id' => $data['_id']],
+            ['$set' => $data]
+        );
+
+        return (bool) $result->getModifiedCount();
+    }
+
+    /**
+     * Removes the given document from the collection.
+     *
+     * @param  mixed $object The object used in the operation.
+     *
+     * @return boolean Success (but always false if write concern is Unacknowledged)
+     */
+    public function delete($object)
+    {
+        if (! $object->_id) {
+            $object->_id = new ObjectID;
+        }
+
+        $data = $this->parseToDocument($object);
+
+        $result = $this->getCollection()->deleteOne(
+            ['_id' => $data['_id']]
+        );
+
+        return (bool) $result->getDeletedCount();
+    }
+
+    /**
+     * Retrieve a database cursor that will return $this->schema->entityClass
+     * objects that upon iteration
+     *
+     * @param  mixed $query MongoDB query to retrieve documents.
+     *
+     * @return \Mongolid\Cursor\Cursor
+     */
+    public function where($query = []): Cursor
+    {
+        $cursor = new Cursor(
+            $this->schema->entityClass,
+            $this->getCollection(),
+            'find',
+            [$this->prepareValueQuery($query)]
+        );
+
+        return $cursor;
+    }
+
+    /**
+     * Retrieve a database cursor that will return all documents as
+     * $this->schema->entityClass objects upon iteration
+     *
+     * @return \Mongolid\Cursor\Cursor
+     */
+    public function all(): Cursor
+    {
+        return $this->where([]);
+    }
+
+    /**
+     * Retrieve one $this->schema->entityClass objects that matches the given
+     * query
+     *
+     * @param  mixed $query MongoDB query to retrieve the document.
+     *
+     * @return mixed First document matching query as an $this->schema->entityClass object
+     */
+    public function first($query)
+    {
+        $document = $this->getCollection()->findOne(
+            $this->prepareValueQuery($query)
+        );
+
+        if (! $document) {
+            return null;
+        }
+
+        $model = new $this->schema->entityClass;
+
+        foreach ($document as $key => $value) {
+            $model->$key = $value;
+        }
+
+        return $model;
+    }
+
+    /**
+     * Parses an object with SchemaMapper and the given Schema
+     *
+     * @param  mixed $object The object to be parsed.
+     *
+     * @return array  Document
+     */
+    protected function parseToDocument($object)
```

Comment:
It can be anything. Like, the user can use the DataMapper to map objects that don't extend anything